cmake/modules/Builduadk.cmake

```diff
@@ -1,5 +1,11 @@
 function(build_uadk)
-    set(configure_cmd ./conf.sh)
+    set(UADK_INSTALL_DIR ${CMAKE_BINARY_DIR}/src/uadk/install)
+    set(UADK_INCLUDE_DIR ${UADK_INSTALL_DIR}/include)
+    set(UADK_LIBRARY_DIR ${UADK_INSTALL_DIR}/lib)
+    set(UADK_WD_LIBRARY ${UADK_LIBRARY_DIR}/libwd.so)
+    set(UADK_WD_COMP_LIBRARY ${UADK_LIBRARY_DIR}/libwd_comp.so)
+    set(configure_cmd env CC=${CMAKE_C_COMPILER} ./configure --prefix=${UADK_INSTALL_DIR})
+    list(APPEND configure_cmd --with-pic --enable-shared --disable-static)
```

Comment:
can this be a static library? that would avoid the extra effort to a) package and install the shared lib with ceph packages, and b) rename things to avoid conflicts with existing libraries installed on the system

Reply:
UADK needs numactl, OpenSSL and other libraries, IMO dynamic library is more lighter-weight and easier.

Reply:
static linkage for uadk doesn't mean it has to statically link openssl and its other deps, does it?

> that would avoid the extra effort to a) package and install the shared lib with ceph packages, and b) rename things to avoid conflicts with existing libraries installed on the system

shared libraries are more complicated for ceph to bundle. if you want to leave it this way, you'll need to rename `libwd.so`/`libwd_comp.so` to `libceph_wd.so`/`libceph_wd_comp.so` to avoid conflicts, and add them to `ceph.spec.in` and `debian/ceph-common.install` so they're installed along with other ceph binaries/libraries

but as we're currently only linking these into a single target (ceph_zlib), we won't actually benefit from a shared library

if/when this library is packaged by our supported distros, we can rely on the shared libraries they provide instead of bundling this way

Reply:
You're right, I forget such rpm matters. Committed a new version, please check~